core/src/com/unciv/models/ruleset/Ruleset.kt

```diff
@@ -12,8 +11,9 @@ import com.unciv.models.ruleset.unit.Promotion
 import com.unciv.models.stats.INamed
 import kotlin.collections.set
 
-class Ruleset {
-    var name=""
+class Ruleset(load: Boolean = true) : JsonParser by JsonParser.Impl() {
```

Comment:
I'm against the Ruleset directly inheriting from JsonParser.
Ruleset is not a *kind of* JsonParser, it's a class with a separate functionality that needs to *use* the functionality of a JsonParser...

Reply:
Well, before me it was a part of ruleset :)
I totally agree with you, but we don't have di like mechanisms to provide this dependency to ruleset
Will check how I can refactor that with minimum efforts

Reply:
We don't need DI, since again this dependency is the only one of its kind, you can just create a JsonParser in the Ruleset.
If at a later date we see that we need to have different json parsers we can add it as a construction parameter to the Ruleset

Reply:
> since again this dependency is the only one of its kind,
I'm not agree, we have a lot of classes which creates multiple times, but it can be created once and used through the whole app cycle
but this is not related to this pr